include/gatekeeper_acl.h

```diff
@@ -79,12 +53,90 @@ add_pkt_ipv6_acl(struct acl_search *acl, struct rte_mbuf *pkt)
 }
 
 static inline int
-process_pkts_ipv6_acl(struct gatekeeper_if *iface, unsigned int lcore,
-	struct acl_search *acl)
+process_pkts_acl(struct gatekeeper_if *iface, unsigned int lcore,
+	struct acl_search *acl, uint16_t proto)
 {
+	struct acl_state *astate;
+	const char *proto_name;
+	bool proto_if_configured;
+
 	if (acl->num == 0)
 		return 0;
-	return process_ipv6_acl(iface, lcore, acl);
+
+	switch (proto) {
+	case ETHER_TYPE_IPv4:
+		astate = &iface->ipv4_acls;
+		proto_name = "IPv4";
+		proto_if_configured = ipv4_if_configured(iface);
+		break;
+	case ETHER_TYPE_IPv6:
```

Comment:
In a regular function, this code is fine. But given that this function is inline and we are trying to help the compiler to get rid of the `switch..case`, drop all current code in the cases `ETHER_TYPE_IPv4` and `ETHER_TYPE_IPv6`, and replace them with `return process_acl(...);`.